astropy/visualization/wcsaxes/ticklabels.py

```diff
@@ -0,0 +1,203 @@
+                    elif np.abs(self.angle[axis][i] - 90.) < 45:
+                        ha = 'center'
+                        va = 'bottom'
+                        dx = 0
+                        dy = - text_size * 1.5
+                    elif np.abs(self.angle[axis][i] - 180.) < 45:
+                        ha = 'left'
+                        va = 'bottom'
+                        dx = text_size * 0.5
+                        dy = - text_size * 0.5
+                    else:
+                        ha = 'center'
+                        va = 'bottom'
+                        dx = 0
+                        dy = text_size * 0.2
+
+                    self.set_position((x + dx, y + dy))
+                    self.set_ha(ha)
+                    self.set_va(va)
+
+                else:
+
+                    # This is the more general code for arbitrarily oriented
+                    # axes
+
+                    # Set initial position and find bounding box
+                    self.set_position((x, y))
+                    bb = super(TickLabels, self).get_window_extent(renderer)
+
+                    # Find width and height, as well as angle at which we
+                    # transition which side of the label we use to anchor the
+                    # label.
+                    width = bb.width
+                    height = bb.height
+                    theta = np.tan(height / width)
+
+                    # Project axis angle onto bounding box
+                    ax = np.cos(np.radians(self.angle[axis][i]))
+                    ay = np.sin(np.radians(self.angle[axis][i]))
+
+                    # Set anchor point for label
+                    if np.abs(self.angle[axis][i]) < 45.:
+                        dx = width
+                        dy = ay * height
+                    elif np.abs(self.angle[axis][i] - 90.) < 45:
+                        dx = ax * width
+                        dy = height
+                    elif np.abs(self.angle[axis][i] - 180.) < 45:
+                        dx = -width
+                        dy = ay * height
+                    else:
+                        dx = ax * width
+                        dy = -height
+
+                    dx *= 0.5
+                    dy *= 0.5
+
+                    # Find normalized vector along axis normal, so as to be
+                    # able to nudge the label away by a constant padding factor
+
+                    dist = np.hypot(dx, dy)
+
+                    ddx = dx / dist
+                    ddy = dy / dist
+
+                    dx += ddx * text_size * self.pad
+                    dy += ddy * text_size * self.pad
+
+                    self.set_position((x - dx, y - dy))
+                    self.set_ha('center')
+                    self.set_va('center')
+
+                bb = super(TickLabels, self).get_window_extent(renderer)
+
+                # TODO: the problem here is that we might get rid of a label
```

Comment:
Should be do-able, not super critical and would like to defer this to a future PR. I can open issues for the TODO/FIXMEs once merged?